SQLite.CodeFirst/SqliteConnectionStringParser.cs

```diff
@@ -26,6 +28,11 @@ internal static class SqliteConnectionStringParser
 
         public static string GetDataSource(string connectionString)
         {
+            if (connectionString.ToLower().Contains(DataDirectoryVariable.ToLower()))
+            {
+                var baseDirectory = AppDomain.CurrentDomain.BaseDirectory + @"\";
```

Comment:
Yes, you are absolutely right.

>This connection string syntax is supported by the SqlClient and OleDb managed providers.
>
>By default, the |DataDirectory| variable will be expanded as follow:
>
> - For applications placed in a directory on the user machine, this will be the >app's (.exe) folder.
>- For apps running under ClickOnce, this will be a special data folder >created by ClickOnce
> - For Web apps, this will be the App_Data folder
>
>Under the hood, the value for |DataDirectory| simply comes from a property on 
>the app domain. It is possible to change that value and override the default behavior by doing this: AppDomain.CurrentDomain.SetData("DataDirectory", newpath)

http://blogs.msdn.com/b/smartclientdata/archive/2005/08/26/456886.aspx

I was able to find .NET Framework Code which does the whole "|DataDirectory|" stuff:
http://referencesource.microsoft.com/#System.Data.Entity/System/Data/EntityClient/DbConnectionOptions.cs,5ed8a8d8098fbe31

Two Options:
1. Call this method with reflection
2. Copy the code

What do you think?